src/storage/src/hummock/compactor/iterator.rs

```diff
@@ -87,6 +92,33 @@ impl SstableStreamIterator {
         sstable_store: SstableStoreRef,
         max_io_retry_times: usize,
     ) -> Self {
+        // filter the block meta with key range
+        let block_metas = filter_block_metas(
+            &block_metas,
+            &existing_table_ids,
+            sstable_info.key_range.clone(),
+        );
+
+        let key_range_left = {
+            let tmp = FullKey::decode(&sstable_info.key_range.left);
+            FullKey::new_with_gap_epoch(
+                tmp.user_key.table_id,
+                TableKey(tmp.user_key.table_key.to_vec()),
+                tmp.epoch_with_gap,
+            )
+        };
+
+        let key_range_right = {
+            let tmp = FullKey::decode(&sstable_info.key_range.right);
+            FullKey::new_with_gap_epoch(
+                tmp.user_key.table_id,
+                TableKey(tmp.user_key.table_key.to_vec()),
+                tmp.epoch_with_gap,
+            )
+        };
```

Comment:
We can use https://github.com/risingwavelabs/risingwave/blob/edb12bb19d815299e19fe4d20a57d3a92103b674/src/storage/hummock_sdk/src/key.rs#L825